contracts/vesting/cw20-vest/src/contract.rs

```diff
@@ -0,0 +1,1219 @@
+                        recipient: info.sender.to_string(),
+                        amount: claimable,
+                    })?,
+                    funds: vec![],
+                };
+                Ok(Response::new()
+                    .add_message(unstake_msg)
+                    .add_message(transfer_msg)
+                    .add_attribute("action", "vest")
+                    .add_attribute("from", info.sender)
+                    .add_attribute("amount", claimable)
+                    .add_attribute("claim_duration", "None"))
+            }
+            Some(duration) => {
+                let outstanding_claims = CLAIMS.query_claims(deps.as_ref(), &info.sender)?.claims;
+                if outstanding_claims.len() >= MAX_CLAIMS as usize {
+                    return Err(ContractError::TooManyClaims {});
+                }
+    
+                CLAIMS.create_claim(
+                    deps.storage,
+                    &info.sender,
+                    claimable,
+                    duration.after(&env.block),
+                )?;
+                Ok(Response::new()
+                    .add_message(unstake_msg)
+                    .add_attribute("action", "vest")
+                    .add_attribute("from", info.sender)
+                    .add_attribute("amount", claimable)
+                    .add_attribute("claim_duration", format!("{}", duration)))
+            }
+        }
+    }
+
+    pub fn claim(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response, ContractError> {
+        if !ACTIVATED.load(deps.storage)? {
+            return Err(ContractError::Unfunded {})
+        }
+
+        let release = CLAIMS.claim_tokens(deps.storage, &info.sender, &env.block, None)?;
+        if release.is_zero() {
+            return Err(ContractError::NothingToClaim {});
+        }
+        let config = CONFIG.load(deps.storage)?;
+        let balance = query_balance(deps.as_ref(), &config.token_address, &env.contract.address)?;
+        if balance < release {
+            let claim_msg = WasmMsg::Execute {
+                contract_addr: config.stake_address.to_string(),
+                msg: to_binary(&cw20_stake::msg::ExecuteMsg::Claim {})?,
+                funds: vec![],
+            };
+            let transfer_msg = cosmwasm_std::WasmMsg::Execute {
+                contract_addr: config.token_address.to_string(),
+                msg: to_binary(&cw20::Cw20ExecuteMsg::Transfer {
+                    recipient: info.sender.to_string(),
+                    amount: release,
+                })?,
+                funds: vec![],
+            };
+            Ok(Response::new()
+                .add_message(claim_msg)
+                .add_message(transfer_msg)
+                .add_attribute("action", "claim")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", release))
+        } else {
+            let transfer_msg = cosmwasm_std::WasmMsg::Execute {
```

Comment:
these two transfer messages are the same. can probably clean up the if statements here to reflect that.

```
let messages = if balance < release {
  let claim_msg = // ...
  vec![claim_msg]
} else {
  vec![]
};
messages.push(transfer_msg);
```